src/app/service/pet/pet.service.go

```diff
@@ -134,8 +143,27 @@ func (s *Service) Create(in *dto.CreatePetRequest) (result *dto.PetResponse, err
 			}
 		}
 	}
-	images := utils.MockImageList(1)[0]
-	createPetResponse := utils.ProtoToDto(res.Pet, images)
+
+	assignRes, assignErr := s.imageService.AssignPet(&dto.AssignPetRequest{
+		Ids:   in.Images,
+		PetId: res.Pet.Id,
+	})
+	if assignErr != nil {
+		return nil, assignErr
+	}
+	if assignRes.Success == false {
+		return nil, &dto.ResponseErr{
+			StatusCode: http.StatusInternalServerError,
+			Message:    constant.InternalErrorMessage,
+			Data:       nil,
+		}
+	}
```

Comment:
Actually, image service's AssignPet doesn't return Success=false when there's error, it'd return nil instead.
So, I think you can remove this. Checking just assignErr != nil should suffice